### ISSUE TITLE
The validation set issue

### ISSUE THREAD

**Issue by yaotao05** [cha15yq/MRC-Crowd#2]
Hello! Thank you for finding my work meaningful for your research. When training on the ShanghaiTechA dataset, how is the validation set selected?

**Comment by cha15yq** [comment]
For SHA and SHB, we didn’t set a validation set.

**Comment by cha15yq** [comment]
I didn't use a validation set for SHA and SHB.

**Comment by yaotao05** [comment]
Is it the epoch with the best MAE or MSE during the training process?

**Comment by yaotao05** [comment]
Is the best result to train in 1000 rounds? The number of rounds I have trained at shanghaitachA is 1500: MAE: 64.99; MSE: 113.07; 102.37180.56 for 1000 rounds on QNRF;

**Comment by yaotao05** [comment]
Is the best result to train in 1000 rounds? The number of rounds I have trained at shanghaitachA is 1500: MAE: 64.99; MSE: 113.07; 102.37，180.56 for 1000 rounds on QNRF;

**Comment by cha15yq** [comment]
I have attached the SHA training log in this repo. Please check that.

**Comment by yaotao05** [comment]
Thank you for your reply. I still have some questions. I saw your log and did you use the data from the test set for verification? What data is still being used? Will using test set data for validation affect the results? Besides, the QNRF pre trained models MAE: 102.37 and MSE: 180.56 (1000 epochs), I have retrained them again now. Is there anything I can advise on?

**Comment by yaotao05** [comment]
Thank you for the author's help. I have just been exposed to deep learning and learned about the application of validation sets. I am currently training on shanghaiA results and have noticed that the learning rate may have been configured incorrectly. May I ask if QNRF and shanghaiA have the same learning rate? Can you publish the QNRF log？

**Comment by cha15yq** [comment]
For QNRF, the learning rate is 2e-5.

**Comment by coleson-CL** [comment]
> I have attached the SHA training log in this repo. Please check that.

Hello, I have reviewed your work and found it very interesting. However, when I was training SHA, the validation results were all NaN, unlike the logs you provided. For example: "Epoch 21 Val, MAE: nan, MSE: nan Cost 0.0 sec." I checked and found that it was because there was no validation set. In the logs you provided, there are validation results. I would like to know what your validation set is?And I would like to know the number of epochs used for training the SHA, SHB, and QNRF datasets, respectively?

**Comment by cha15yq** [comment]
For SHA,SHB, I used the test set for validation. For QNRF, the validation set is the same with that provided in Bayesian Loss

**Comment by coleson-CL** [comment]
> For SHA,SHB, I used the test set for validation. For QNRF, the validation set is the same with that provided in Bayesian Loss

Thank you for your prompt response. Additionally, I would like to know how many epochs you used for each dataset. Was it 200 for all of them?

**Comment by cha15yq** [comment]
For QNRF, try 500 epochs.

**Comment by coleson-CL** [comment]
> For QNRF, try 500 epochs.
> […](#)
Hello, I recently tried to reproduce your code using the QNRF dataset but found that I couldn't achieve the results reported in the paper. I suspect there might be significant differences in parameter settings. Could you please provide the training logs for QNRF as a reference? Thank you.

**Comment by cha15yq** [comment]
I don't have these logs now since I have moved to a  new instituion. However the training script is modified based on experiments on QNRF. Some hints: try more seeds and also save some models with good MAE performance on validation set. I found sometimes a single image will greatly influence the overall performance for crowd counting task, so don't only trust on the best MAE+MSE model.
